src/App.tsx

```diff
@@ -1,26 +1,181 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { USER_ID, getTodos, deleteTodo, addTodo } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { Error } from './types/Error';
+import { Filter } from './types/Filter';
+import { getFilteredTodos } from './helpers';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoFooter } from './components/TodoFooter';
+import { ErrorNotification } from './components/ErrorNotification';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { TodoItem } from './components/TodoItem';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [currentError, setCurrentError] = useState<Error | null>(null);
+  const [currentFilter, setCurrentFilter] = useState<Filter>(Filter.All);
+  const [deleteTodoId, setDeleteTodoId] = useState<Todo['id'] | null>(null);
+  const [addTodoTitle, setAddTodoTitle] = useState<string>('');
+  const [tempTodo, setTempTodo] = useState<null | Todo>(null);
+  const [isNewTodoLoading, setIsNewTodoLoading] = useState<boolean>(false);
+
+  const hasCompletedTodos = todos.some((todo: Todo) => todo.completed);
+
+  const activeTodosAmount = todos.filter(
+    (todo: Todo) => !todo.completed,
+  ).length;
+
+  const newTodoInput = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    setCurrentError(null);
+
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setCurrentError(Error.CannotLoad);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (currentError) {
+      newTodoInput.current?.focus();
+
+      setTimeout(() => setCurrentError(null), 3000);
+    }
+  }, [currentError]);
+
+  useEffect(() => {
+    newTodoInput.current?.focus();
+  }, [todos, newTodoInput]);
+
+  useEffect(() => {
+    if (deleteTodoId) {
+      setCurrentError(null);
+
+      deleteTodo(deleteTodoId)
+        .then(() => {
+          getTodos()
```

Comment:
This API call is pretty much redundant: you never want to send an API call to 'get the whole list of items once again'  after delete/put/post request. You backend will be overloaded and users with slow connection will be irritated. 

If it was successful, knowing the item properties, you can easily update your UI. 

Also, notice how quickly your code becomes flooded with .catch statements. Not cool.